### PR TITLE
2279 feat: implement estimate gas

### LINKED ISSUE #2279: Implement `estimateGas()` in the websocket server.
### Problem

When using websockets some JSON RPC methods are not available. `estimateGas()` is one of them.

### Solution

Enhance the outer switch, case, statement in the webSocketServer.ts to call the `estimateGas()` method in the Relay.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ebadiere**
Implemented `eth_estimateGas`

**Related issue(s)**:

Fixes #2279

### CODE REVIEW

File: packages/server/tests/acceptance/ws/estimateGas.spec.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  });
+
+  it('@release should execute "eth_estimateGas" for contract call, using a websocket provider', async function () {
+    const estimatedGas = await wsProvider.estimateGas({
+      to: `0x${basicContract.contractId.toSolidityAddress()}`,
+      data: BASIC_CONTRACT_PING_CALL_DATA,
+    });
+
+    // handle deviation in gas price
+    expect(Number(estimatedGas)).to.be.lessThan(currentPrice * (1 + gasPriceDeviation));
```
Comment (quiet-node) [inline packages/server/tests/acceptance/ws/estimateGas.spec.ts:67]: Maybe add an explaination on this logic?

File: packages/server/tests/acceptance/ws/estimateGas.spec.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    accounts[0] = await servicesNode.createAliasAccount(100, relay.provider, requestId);
+    basicContract = await servicesNode.deployContract(basicContractJson);
+    wsProvider = await new ethers.WebSocketProvider(WS_RELAY_URL);
+    provider = new ethers.JsonRpcProvider(RELAY_URL);
+
+    webSocket = new WebSocket(WS_RELAY_URL);
+    currentPrice = await relay.gasPrice(requestId);
+    expectedGas = parseInt(PING_CALL_ESTIMATED_GAS, 16);
+
+    gasPriceDeviation = parseFloat(expectedGas.toString() ?? '0.2');
```
Comment (quiet-node) [inline packages/server/tests/acceptance/ws/estimateGas.spec.ts:57]: Maybe add an explaination on this logic?

**Review by quiet-node — COMMENTED** [review]
LG but some nits and questions

File: packages/ws-server/src/controllers/eth_estimateGas.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ */
+
+import { sendToClient } from '../utils/utils';
+import { Relay } from '@hashgraph/json-rpc-relay';
+import { validateParamsLength } from '../utils/validators';
+import { handleSendingTransactionRequests } from './helpers';
+
+/**
+ * Handles the "eth_getTransactionByHash" method request by retrieving transaction details from the Hedera network.
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply packages/ws-server/src/controllers/eth_estimateGas.ts:27]: Done.

File: packages/ws-server/src/controllers/eth_estimateGas.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    logger,
+    sendToClient,
+    1,
+    socketIdPrefix,
+    requestIdPrefix,
+    connectionIdPrefix,
+  );
+
+  logger.info(
+    `${connectionIdPrefix} ${requestIdPrefix} ${socketIdPrefix}: Retrieving transaction info with txn=${TXN} for tag=${TAG}`,
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply packages/ws-server/src/controllers/eth_estimateGas.ts:69]: Done.

File: packages/server/tests/acceptance/ws/estimateGas.spec.ts (hunk 1 above)
Comment (ebadiere) [reply packages/server/tests/acceptance/ws/estimateGas.spec.ts:67]: Done.

File: packages/server/tests/acceptance/ws/estimateGas.spec.ts (hunk 2 above)
Comment (ebadiere) [reply packages/server/tests/acceptance/ws/estimateGas.spec.ts:57]: Done.

**Review by quiet-node — APPROVED** [review]
LGTM

**Pull request merged**